### PR TITLE
fix: deprecate use of `renovate.json` as preset

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Deprecate the use of `renovate.json` as a preset in our docs
- Add `logger.info` deprecation message

## Context:

Based on https://github.com/renovatebot/renovate/discussions/10671#discussioncomment-944174

I think we still need to deprecate this in our code somehow?

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/usage/config-presets.md [hunk 1]
```diff
@@ -124,12 +129,12 @@ Or if you think your preset would be valuable for others, please contribute a PR
...
 
 It is also possible to host your preset config using just a regular GitHub repository and without needing to publish it to npmjs.
-In such cases Renovate will simply look for a `renovate.json` file in the default branch, e.g. `main`.
+In such cases Renovate will simply look for a `default.json` file in the default branch, e.g. `main`.
 
 To host your preset config on GitHub:
 
 - Create a new repository. Normally you'd call it `renovate-config` but it can be named anything
-- Add configuration files to this new repo for any presets you want to share. For the default preset, `default.json` will be checked first and then `renovate.json`. For named presets, `<preset-name>.json` will be loaded. For example, loading preset `library` would load `library.json`. No other files are necessary.
+- Add configuration files to this new repo for any presets you want to share. For the default preset, only `default.json` will be checked. For named presets, `<preset-name>.json` will be loaded. For example, loading preset `library` would load `library.json`. No other files are necessary.
```
Comment (viceice) [inline docs/usage/config-presets.md:137]:
```suggestion
- Add configuration files to this new repo for any presets you want to share. For the default preset, `default.json` will be checked. For named presets, `<preset-name>.json` will be loaded. For example, loading preset `library` would load `library.json`. No other files are necessary.
```

I think we don't need to be so explicit.

File: docs/usage/config-presets.md [hunk 2]
```diff
@@ -212,6 +217,9 @@ In that case, you can choose between publishing your preset config package as `@
 Let's assume you choose `renovate-config-fastcore` as the package name.
 
 You then need to publish the `renovate-config-fastcore` package where the `package.json` contains the field `renovate-config` and then put your config under the field `default`.
+
+<!-- TODO: do we need to change the renovate-config-fastcore to use default.json as well? -->
+
```
Comment (viceice) [inline docs/usage/config-presets.md:222]:
```suggestion
```

no as there the `package.json` config ist used for npm based presets.

**Review by viceice — COMMENTED** [review]
I like this docs only deprecation for now. So new users use the `default.json` but existing users are not yet forced to do anything.

We can add a warning to the code in a later pr to have a log message.

Maybe we can exend this message to include a deprecation fallback for now

https://github.com/renovatebot/renovate/blob/9ca604b3453ca55795bd10d3fcad4475c3c8c7cb/lib/config/presets/util.ts#L37-L37

File: lib/config/presets/util.ts [hunk 3]
```diff
@@ -35,6 +35,9 @@ export async function fetchPreset({
         throw err;
       }
       logger.debug('default.json preset not found - trying renovate.json');
+      logger.info(
+        'Using a renovate.json file as a preset is deprecated, please use a default.json file instead.'
+      );
```
Comment (HonkingGoose) [inline lib/config/presets/util.ts:40]: I don't think the new `logger.info` message is actually triggering correctly. I think I need to move it down a bit?

Comment (viceice) [reply lib/config/presets/util.ts:40]: i think info is ok for now, but message is a little bit misleading. Maybe this ?
```suggestion
      logger.info(
        'Fallback to renovate.json file as a preset is deprecated, please use a default.json file instead.'
      );
```

**Review by viceice — COMMENTED** [review]
Updated comment 😉

**Review by HonkingGoose — COMMENTED** [review]
> you can still use `renovate.json` as explicit preset, eg `local>org/repo:renovate` .

Are the tables correct, or do I need to adjust them?

Also are we sure the `logger.info` message is in the correct spot in the code?

File: lib/config/presets/util.ts [hunk 4]
```diff
@@ -35,6 +35,9 @@ export async function fetchPreset({
         throw err;
       }
       logger.debug('default.json preset not found - trying renovate.json');
```
Comment (viceice) [inline lib/config/presets/util.ts:37]:
```suggestion
```
We don't need both log messages.

**Review by viceice — COMMENTED** [review]
Docs are ok and it's also right code place

**Review by viceice — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**